tests/views/test_blog_pages.py

```diff
@@ -0,0 +1,87 @@
+from tests import PMGLiveServerTestCase
+from tests.fixtures import dbfixture, PostData
+import flask
+
+from pmg.models import db, Post
+
+
+class TestBlogPages(PMGLiveServerTestCase):
+    def setUp(self):
+        super(TestBlogPages, self).setUp()
+
+        self.fx = dbfixture.data(PostData)
+        self.fx.setup()
+
+    def tearDown(self):
+        self.fx.teardown()
+        super(TestBlogPages, self).tearDown()
+
+    def test_blog_post_page(self):
+        """
+        Test blog post page (http://pmg.test:5000/blog/<slug>)
+        """
+        post = self.fx.PostData.the_week_ahead
+        self.get_page_contents("http://pmg.test:5000/blog/%s/" % post.slug)
+        self.assertIn(post.title, self.html)
+        self.assertIn(post.body[0:100], self.html)
+        self.assertIn('That week in Parliament', self.html)
+
+    def test_blog_listings_page(self):
+        """
+        Test blog listings page (http://pmg.test:5000/blog)
+        """
+        self.get_page_contents("http://pmg.test:5000/blog")
+        self.contains_template_text()
+        self.contains_posts(Post.query.all())
+        self.contains_archive()
+
+    def test_blog_listings_page_with_filter(self):
+        """
+        Test blog listings page with a filter 
+        (http://pmg.test:5000/blog?filter[month]=<month>&filter[year]=<year>).
+        """
+        month = 'February'
+        year = 2019
+        self.get_page_contents(
+            "http://pmg.test:5000/blog?filter[month]=%s&filter[year]=%d" %
+            (month, year))
+        self.contains_template_text()
+        self.contains_posts([
+            self.fx.PostData.first_term_review,
+            self.fx.PostData.brief_explainer,
+        ])
+        self.doesnt_contain_posts([
+            self.fx.PostData.the_week_ahead,
+            self.fx.PostData.government_priorities,
+        ])
+        self.contains_archive()
+
+    def contains_template_text(self):
+        self.assertIn('That week in Parliament', self.html)
+        self.assertIn('About this blog', self.html)
+        self.assertIn('Blog Archive', self.html)
+
+    def contains_posts(self, posts):
+        for post in posts:
+            self.contains_blog_post(post)
+
+    def contains_blog_post(self, post):
+        self.assertIn(flask.escape(post.title), self.html)
```

Comment:
The comment on hardcoding doesn't make so much sense here, but perhaps the arguments to this and contains_posts could be a list of strings and IDs or something